File: VB/ReplaceDefaultActionInListView.Module/AddressBookRecord.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo
Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace ReplaceDefaultActionInListView.Module
    <DefaultClassOptions, ImageName("BO_Contact")> _
    Public Class AddressBookRecord
        Inherits BaseObject
        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Private _name As String
        Public Property Name() As String
            Get
                Return _name
            End Get
            Set(ByVal value As String)
                SetPropertyValue("Name", _name, value)
            End Set
        End Property
        Private _email As String
        Public Property Email() As String
            Get
                Return _email
            End Get
            Set(ByVal value As String)
                SetPropertyValue("Email", _email, value)
            End Set
        End Property
        Private _phoneNumber As String
        Public Property PhoneNumber() As String
            Get
                Return _phoneNumber
            End Get
            Set(ByVal value As String)
                SetPropertyValue("PhoneNumber", _phoneNumber, value)
            End Set
        End Property
    End Class
End Namespace

File: CS/ReplaceDefaultActionInListView.Module/AddressBookRecord.cs
using System;
using DevExpress.Xpo;
using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace ReplaceDefaultActionInListView.Module {
    [DefaultClassOptions, ImageName("BO_Contact")]
    public class AddressBookRecord : BaseObject {
        public AddressBookRecord(Session session) : base(session) { }
        private string name;
        public string Name {
            get { return name; }
            set { SetPropertyValue("Name", ref name, value); }
        }
        private string email;
        public string Email {
            get { return email; }
            set { SetPropertyValue("Email", ref email, value); }
        }
        private string phoneNumber;
        public string PhoneNumber {
            get { return phoneNumber; }
            set { SetPropertyValue("PhoneNumber", ref phoneNumber, value); }
        }
    }
}
